src/components/ManagerPage/FirmwareUpdate.js

```diff
@@ -89,31 +92,49 @@ class FirmwareUpdate extends PureComponent<Props, State> {
   }
 
   render() {
-    const { t, ...props } = this.props
+    const { t, infos } = this.props
     const { latestFirmware } = this.state
 
-    if (!latestFirmware) {
-      return null
-    }
-
     return (
-      <Box flow={4} {...props}>
-        <Card flow={2} {...props}>
-          <Box horizontal align="center" flow={2}>
-            <Text ff="Open Sans|Regular" fontSize={4}>{`${t('manager:latestFirmware')}: ${
-              latestFirmware.name
-            }`}</Text>
-            <Button outline onClick={this.installFirmware(latestFirmware)}>
-              {t('manager:install')}
-            </Button>
+      <Card px={4} py={25}>
+        <Box horizontal align="center" flow={2}>
+          <Box color="dark">
+            <NanoS size={30} />
+          </Box>
+          <Box>
+            <Box horizontal align="center">
+              <Text ff="Open Sans|SemiBold" fontSize={4} color="dark">
+                Ledger Nano S
+              </Text>
+              <Box color="wallet" style={{ marginLeft: 10 }}>
+                <CheckFull size={13} color="wallet" />
+              </Box>
+            </Box>
+            <Text ff="Open Sans|SemiBold" fontSize={2}>
+              Firmware {infos.version}
+            </Text>
           </Box>
-          <Box
-            fontSize={3}
-            style={{ whiteSpace: 'pre' }}
-            dangerouslySetInnerHTML={{ __html: latestFirmware.notes }}
-          />
-        </Card>
-      </Box>
+          {latestFirmware && (
+            <Fragment>
+              <Text
+                ff="Open Sans|Regular"
+                fontSize={4}
+                style={{ marginLeft: 'auto', marginRight: 15 }}
+              >
+                {t('manager:latestFirmware', { version: latestFirmware.name })}
+              </Text>
+              <Button primary onClick={this.installFirmware(latestFirmware)}>
+                {t('manager:installFirmware')}
+              </Button>
+            </Fragment>
+          )}
```

Comment:
it would be great if you could split things into more components.
I've been started doing this in our project, it helps performance, readability and maintainability ;)

for instance here, it could be a component that takes a `firmware` prop (and `device` prop) so you don't need to create a function each time.

Reply:
I fixed it, please tell me if this is what you were expecting @gre

Reply:
mmh i think `UpdateButton` could be in a separate file implmented as a Component and that have a local method  `installFirmware =  async () => {` so that relieve a bit the codebase of this component that don't need `  installFirmware = (firmware: FirmwareInfos) => async () => {`

even more simply, you can provably directly access this.state.latestFirmware so you don't need to do this nested function

Reply:
Okay, fixes are done @gre